src/utils/solidity-keccak256.ts

```diff
@@ -1,3 +1,81 @@
+import { tinyBig } from "../shared/tiny-big/tiny-big";
+import { Buffer } from "buffer";
+import { Keccak } from "sha3";
+
+// Importing from ethers for the time being until an optimized essential-eth option can be made
+import { arrayify, zeroPad, concat, hexlify } from "./bytes";
+import { logger } from "../logger/logger";
+
+const regexBytes = new RegExp("^bytes([0-9]+)$");
+const regexNumber = new RegExp("^(u?int)([0-9]*)$");
+const regexArray = new RegExp("^(.*)\\[([0-9]*)\\]$");
+
+const Zeros = "0000000000000000000000000000000000000000000000000000000000000000";
+
+function pack(type: string, value: any, isArray?: boolean): Uint8Array {
+  switch(type) {
+      case "address":
+          if (isArray) { return zeroPad(value, 32); }
+          return arrayify(value);
+      case "string":
+          return Buffer.from(value);
+      case "bytes":
+          return arrayify(value);
+      case "bool":
+          value = (value ? "0x01": "0x00");
+          if (isArray) { return zeroPad(value, 32); }
+          return arrayify(value);
+  }
+
+  let match =  type.match(regexNumber);
+  if (match) {
+      //let signed = (match[1] === "int")
+      let size = parseInt(match[2] || "256")
+
+      if ((match[2] && String(size) !== match[2]) || (size % 8 !== 0) || size === 0 || size > 256) {
+          logger.throwArgumentError("invalid number type", "type", type)
+      }
+
+      if (isArray) { size = 256; }
+
+      // value = BigNumber.from(value).toTwos(size);
+      value = tinyBig(value)
```

Comment:
I had trouble figuring out exactly what the `toTwos` function does, but it should be reexamined before this PR is merged

Reply:
Nope, that's what I'm working on now. Definitely not ready to be merged, just wanted to get what I had up here